src/mbgl/style/conversion/source.cpp

```diff
@@ -86,8 +86,25 @@ static optional<std::unique_ptr<Source>> convertVectorSource(const std::string&
     if (!urlOrTileset) {
         return nullopt;
     }
-
-    return { std::make_unique<VectorSource>(id, std::move(*urlOrTileset)) };
+    auto maxzoomValue = objectMember(value, "maxzoom");
+    optional<float> maxzoom;
+    if (maxzoomValue) {
+        maxzoom = toNumber(*maxzoomValue);
+        if (!maxzoom || *maxzoom < 0 || *maxzoom > std::numeric_limits<uint8_t>::max()) {
```

Comment:
Should it be capped to 22?

Reply:
Locally agreed with that there is possibility that the customer would provide higher zoom level maps. Leave it as it is for now.